### ISSUE TITLE
DXC error when writing to a float3x4 vertex output within a mesh shader 

### ISSUE THREAD

**Issue by belzen** [microsoft/DirectXShaderCompiler#3341]
Hello,
When compiling a mesh shader that has a float3x4 or float3x3 as a vertex output, DXC returns:
> warning: no output provided for debug - embedding PDB in shader container.  Use -Qembed_debug to silence this warning.
> error: validation errors
> Invalid record
> Validation failed.

This does not occur if the float3x4 is never written to, however.  It also seems to only be an issue with mesh shaders.  Attempting to write to individual elements of the float3x4 also causes DXC to silently crash or do nothing.

The following shader code reproduces the issue with both the October release and the latest build's artifacts:
```
struct VertexOutput
{
		float4 pos : SEM_POSITION_OUT;
		float3x4 test : TEXCOORD0;
};

[NumThreads(64, 1, 1)]
[OutputTopology("triangle")]
void main(out indices uint3 tris[64], out vertices VertexOutput verts[192])
{
	verts[0].test = float3x4(float4(0,1,2,0), float4(3,4,5,0), float4(6,7,8,0));
	SetMeshOutputCounts(0, 0);
}
```

Tested using:
> dxc -Zi -T ms_6_5 test.hlsl`

Interestingly, I accidentally found that if I remove the 'dxil.dll' from the folder, there are actual errors that get written out related to the float3x4.  In case they're useful:

> warning: no output provided for debug - embedding PDB in shader container.  Use -Qembed_debug to silence this warning.
> warning: DXIL.dll not found.  Resulting DXIL will not be signed for use in release environments.
> 
> error: validation errors
> Incorrect number of arguments passed to called function!
>   call void @dx.op.storeVertexOutput.f32(i32 171, i32 1, i32 0, i8 0, float 0.000000e+00)
> Incorrect number of arguments passed to called function!
>   call void @dx.op.storeVertexOutput.f32(i32 171, i32 1, i32 0, i8 1, float 3.000000e+00)
> Incorrect number of arguments passed to called function!
>   call void @dx.op.storeVertexOutput.f32(i32 171, i32 1, i32 0, i8 2, float 6.000000e+00)
> Incorrect number of arguments passed to called function!
>   call void @dx.op.storeVertexOutput.f32(i32 171, i32 1, i32 1, i8 0, float 1.000000e+00)
> Incorrect number of arguments passed to called function!
>   call void @dx.op.storeVertexOutput.f32(i32 171, i32 1, i32 1, i8 1, float 4.000000e+00)
> Incorrect number of arguments passed to called function!
>   call void @dx.op.storeVertexOutput.f32(i32 171, i32 1, i32 1, i8 2, float 7.000000e+00)
> Incorrect number of arguments passed to called function!
>   call void @dx.op.storeVertexOutput.f32(i32 171, i32 1, i32 2, i8 0, float 2.000000e+00)
> Incorrect number of arguments passed to called function!
>   call void @dx.op.storeVertexOutput.f32(i32 171, i32 1, i32 2, i8 1, float 5.000000e+00)
> Incorrect number of arguments passed to called function!
>   call void @dx.op.storeVertexOutput.f32(i32 171, i32 1, i32 2, i8 2, float 8.000000e+00)
> Incorrect number of arguments passed to called function!
>   call void @dx.op.storeVertexOutput.f32(i32 171, i32 1, i32 3, i8 0, float 0.000000e+00)
> Incorrect number of arguments passed to called function!
>   call void @dx.op.storeVertexOutput.f32(i32 171, i32 1, i32 3, i8 1, float 0.000000e+00)
> Incorrect number of arguments passed to called function!
>   call void @dx.op.storeVertexOutput.f32(i32 171, i32 1, i32 3, i8 2, float 0.000000e+00)
> Module bitcode is invalid
> 
> Validation failed.
